src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -422,6 +424,33 @@ private static void ThrowInvalidDocumentId(string id)
             }
             else
             {
+                // check for attachments with duplicate name
+                if ((flags & DocumentFlags.Resolved) == DocumentFlags.Resolved)
+                {
+                    if (metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray currentDocumentAttachments))
+                    {
+                        // actualAttachments can contain attachments of current document which were directly put, we want to check if there are attachment with duplicate name
+                        var attachmentsChanged = false;
+                        foreach (BlittableJsonReaderObject bjro in currentDocumentAttachments)
+                        {
+                            var currentAttachment = JsonDeserializationClient.AttachmentName(bjro);
+
+                            if (actualAttachments.Items.Count(o =>
+                                o is DynamicJsonValue attachment && attachment[nameof(AttachmentName.Name)] is LazyStringValue lzv && lzv == currentAttachment.Name) > 1)
+                            {
+                                // rename currentAttachment which is already saved in storage
+                                _documentsStorage.AttachmentsStorage.RenameAttachment(
```

Comment:
What if we have three duplicates?

Reply:
The only chance for this is when the doc already has attachments with duplicate names (got conflicted before this PR merged) and gets into conflict again after this PR merged. For normal doc, after this PR there is always going to be max 2 duplicates.

We can treat this issue by treating here the > 2 duplicates case.